src/utils/crypto.ts

```diff
@@ -0,0 +1,96 @@
+import * as crypto from "crypto";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+
+/**
+ * Returns hashed input based on sha256 algo
+ */
+export const toHash = (s: string): NonEmptyString =>
+  crypto
+    .createHash("sha256")
+    .update(s)
+    .digest("hex") as NonEmptyString;
+
+/**
+ * Kindly imported from node-http-signature (https://github.com/eBay/digital-signature-nodejs-sdk/blob/main/lib/helpers/digest-helper.ts)
+ */
+
+export const CONTENT_DIGEST_CONSTANTS = {
+  BASE64: "base64",
+  COLON: ":",
+  CONTENT_DIGEST_SHA256: "sha-256=:",
+  CONTENT_DIGEST_SHA512: "sha-512=:",
+  SHA_256: "sha256",
+  SHA_512: "sha512"
+} as const;
+
+/**
+ * Generates the 'Content-Digest' header value for the input payload.
+ *
+ * @param {Buffer} payload The request payload.
+ * @param {string} cipher The algorithm used to calculate the digest.
+ * @returns {string} contentDigest The 'Content-Digest' header value.
+ */
+export const generateDigestHeader = (
+  payload: Buffer | string,
+  cipher:
+    | typeof CONTENT_DIGEST_CONSTANTS.SHA_256
+    | typeof CONTENT_DIGEST_CONSTANTS.SHA_512
+): string => {
+  // Validate the input payload
+  if (!payload) {
+    return "";
+  }
+
+  // Calculate the SHA-256 digest
+  const hash = crypto
+    .createHash(cipher)
+    .update(payload)
+    .digest(CONTENT_DIGEST_CONSTANTS.BASE64);
+
+  const algo: string =
+    cipher === CONTENT_DIGEST_CONSTANTS.SHA_512
+      ? CONTENT_DIGEST_CONSTANTS.CONTENT_DIGEST_SHA512
+      : CONTENT_DIGEST_CONSTANTS.CONTENT_DIGEST_SHA256;
+
+  return algo + hash + CONTENT_DIGEST_CONSTANTS.COLON;
+};
+
+/**
+ * Validates the 'Content-Digest' header value.
+ *
+ * @param {string} contentDigestHeader The Content-Digest header value.
+ * @param {Buffer} body The HTTP request body.
+ * @throws {Error} If the Content-Digest header value is invalid.
+ */
+export const validateDigestHeader = (
+  contentDigestHeader: string,
+  body: Buffer | string
+): void => {
+  if (!contentDigestHeader) {
+    throw new Error("Content-Digest header missing");
+  }
+
+  // Validate
+  const contentDigestPattern = new RegExp("(.+)=:(.+):");
```

Comment:
We can reduce the regex vulnerability with this
```suggestion
  const contentDigestPattern = new RegExp("(sha-256|sha-512)=:(.+):");
```